wled00/bus_manager.cpp

```diff
@@ -162,7 +161,12 @@ void IRAM_ATTR BusDigital::setPixelColor(uint16_t pix, uint32_t c) {
       case 2: c = RGBW32(R(cOld), G(cOld), W(c)   , 0); break;
     }
   }
-  PolyBus::setPixelColor(_busPtr, _iType, pix, c, co);
+  
+  uint8_t ww = W(c), cw = 0;
+  if (_type == TYPE_FW1906)
+    calculateCCT(c, ww, cw);
+
+  PolyBus::setPixelColor(_busPtr, _iType, pix, R(c), G(c), B(c), ww, cw, co);
```

Comment:
Hi, 
setPixelColor() is on the critical path for performance.

Is it strictly necessary to extend the polyBus interface to explicitly require color components (r g,b,w,)  instead keeping them in uint32_t and just adding the new byte for cw?

If I understand your code correctly, we would require - for all kinds of busses - to split RGBW into  `R(c), G(c), B(c), W(c), 0`, and later re-assemble  them into uint32_t? This adds around 20 extra operations just to go back and forth between color formats. It might seem like a "small price", but if applied to a few thousand LEDs at 100 fps, it might become a noticeable slowdown.

Reply:
Hi,
The split is not required. I can change the interface. But in PolyBus::setPixelColor the uint32 is split for 4 uint8 local variables anyway, and this operation is removed. I don't want to guess witch version is faster.

Reply:
Thanks for the clarification :-) so before changing, @blazoncek what do you think, which option would be more efficient?

Reply:
I would leave API as is and add a static variable to hold CCT information for PolyBus.